src/graphEntry.ts

```diff
@@ -193,8 +218,15 @@ export default class GraphEntry {
     let startHistory = new Date(start);
     if (this._config.group_by.func !== 'raw') {
       const range = end.getTime() - start.getTime();
-      const nbBuckets = Math.floor(range / this._groupByDurationMs) + (range % this._groupByDurationMs > 0 ? 1 : 0);
+      const monthMode = /\bmonth\b/i.test(this._config?.group_by?.duration);
+      const nbBuckets = monthMode
+        ? (monthsBetween(startOfMonth(start.getTime()), startOfMonth(end.getTime())))
+        : (Math.floor(range / this._groupByDurationMs) +
+          ((range % this._groupByDurationMs) > 0 ? 1 : 0));  
       startHistory = new Date(end.getTime() - (nbBuckets + 1) * this._groupByDurationMs);
+      this._buckets = monthMode
+        ? Array.from({ length: nbBuckets + 1 }, (_,_i) => addMonths(startOfMonth(start.getTime()), _i))
+        : Array.from({ length: nbBuckets + 2 }, (_,_i) => end.getTime() - _i * this._groupByDurationMs).reverse();    
```

Comment:
Why change the default behavior and move it here instead of keeping it in `_dataBucketer`?